pkg/server/autoconfig/auto_config.go

```diff
@@ -79,13 +109,25 @@ func (r *autoConfigRunner) Resume(ctx context.Context, execCtx interface{}) erro
 		case <-waitForEnvChange:
 		}
 
-		r.job.MarkIdle(false)
 		for _, envID := range provider.ActiveEnvironments() {
+			// Check if the rate limit has changed. If it has, make a new limiter.
+			if curLimit := autoConfigMaxNewRunnersPerSecond.Get(&execCfg.Settings.SV); curLimit != prevLimit || limiter == nil {
+				limiter = rate.NewLimiter(rate.Limit(curLimit), 1)
+				prevLimit = curLimit
+			}
+			// Wait according to the configured rate limit.
+			limiter.Wait(ctx)
+
+			// Ensure that the idleness is not changed while we call
```

Comment:
On the first iteration of this loop if we are blocking on Wait then our `idle` bit will default to false thereby preventing server shutdown. Is this okay or should we mark the job as idle before we call Wait? The comment above MarkIdle also indicates that this bit shouldn't be toggled more than twice a minute because the flip is logged and may be persisted to job state in the future. Do you think moving the flip inside the loop goes against what the comment suggests?